openedx/core/djangoapps/edx_global_analytics/utils.py

```diff
@@ -0,0 +1,142 @@
+"""
+Helpers for the edX global analytics application.
+"""
+
+import datetime
+import calendar
+
+from django.core.cache import cache
+
+from django.db.models import Count
+from django.db.models import Q
+from student.models import UserProfile
+
+
+def fetch_instance_information(name_to_cache, query_type, activity_period, cache_timeout=None):
+    """
+    Calculates instance information corresponding for particular period as like as previous calendar day and
+    statistics type as like as students per country after cached if needed.
+    """
+    period_start, period_end = activity_period
+
+    statistics_queries = {
+        'active_students_amount': UserProfile.objects.exclude(
+            Q(user__last_login=None) | Q(user__is_active=False)
+        ).filter(user__last_login__gte=period_start, user__last_login__lt=period_end).count(),
+
+        'students_per_country': dict(UserProfile.objects.exclude(
+            Q(user__last_login=None) | Q(user__is_active=False)
+        ).filter(user__last_login__gte=period_start, user__last_login__lt=period_end).values(
+            'country'
+        ).annotate(count=Count('country')).values_list('country', 'count'))
```

Comment:
Change indentation according to [Open edX Python code style guide](https://edx.readthedocs.io/projects/edx-developer-guide/en/latest/style_guides/python_guidelines.html).